### PR TITLE
feat(analyzer): detect duplicate storage map prefixes

### PR DESCRIPTION
**Pull request by Jim8y**
## Summary
- add a Roslyn analyzer that flags duplicate constant `StorageMap` / `LocalStorageMap` prefixes declared in the same contract type
- normalize byte, string, and byte-array prefixes so obvious namespace collisions are surfaced early
- add focused analyzer unit tests for duplicate byte and string prefixes plus non-collision cases

## Context
- first implementation slice for `Security: Storage key collision detection` from #1514
- based on the latest `master-n3`

## Test Plan
- dotnet test tests/Neo.SmartContract.Analyzer.UnitTests/Neo.SmartContract.Analyzer.UnitTests.csproj --filter FullyQualifiedName~StorageKeyCollisionAnalyzerUnitTests

### CODE REVIEW

File: src/Neo.SmartContract.Analyzer/StorageKeyCollisionAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            Dictionary<string, PrefixUsage> seenPrefixes = new(StringComparer.Ordinal);
+            foreach (IFieldSymbol field in typeSymbol.GetMembers().OfType<IFieldSymbol>())
+            {
+                if (field.DeclaringSyntaxReferences.Length == 0)
+                    continue;
+
+                if (field.DeclaringSyntaxReferences[0].GetSyntax(context.CancellationToken) is not VariableDeclaratorSyntax declarator)
+                    continue;
+
+                if (declarator.Initializer?.Value is not BaseObjectCreationExpressionSyntax creation)
```
Comment (ajara87) [inline src/Neo.SmartContract.Analyzer/StorageKeyCollisionAnalyzer.cs]: I think this currently only analyzes direct object creation (new ...) but not assignments like XXX = CreateXXXX();

Comment (Jim8y) [reply src/Neo.SmartContract.Analyzer/StorageKeyCollisionAnalyzer.cs]: Fixed in ac37f2f8. The analyzer now follows parameterless factory initializers like `Owners = CreateOwners();` and resolves the returned `StorageMap`/`LocalStorageMap` prefix before checking for collisions. Added a regression test covering both block-bodied and expression-bodied factory methods.

**Comment by shargon** [comment]
Love it!!! Thabks @Jim8y

**Review by shargon — APPROVED** [review]

**Review by ajara87 — APPROVED** [review]
